modules/mux/targets/riscv/source/kernel.cpp

```diff
@@ -50,10 +47,19 @@ cargo::expected<riscv::kernel_s *, mux_result_t> kernel_s::create(
     variant.variant_name = meta.kernel_name;
     variant.sub_group_size = meta.sub_group_size.getKnownMinValue();
     if (meta.sub_group_size.isScalable()) {
+      // Assert non-zero as sub_group_size is unknowable without vlen != 0
+      assert(
+          hal_device_info->vlen &&
+          "vlen must be known at runtime to calculate scalable subgroup width");
       variant.sub_group_size *= real_vscale;
     }
     variant.min_work_width = meta.min_work_item_factor.getKnownMinValue();
+    variant.min_work_width = meta.min_work_item_factor.getKnownMinValue();
     if (meta.min_work_item_factor.isScalable()) {
+      // Assert non-zero as sub_group_size is unknowable without vlen != 0
+      assert(
+          hal_device_info->vlen &&
+          "vlen must be known at runtime to calculate scalable subgroup width");
```

Comment:
There is another use of `real_vscale` below that also needs the same assert. (Is there perhaps a way we can write that assert only once? Maybe make `real_scale` a lambda that includes the assert?)

Reply:
i've tried something similar